File: tests/fixtures/anti-patterns/csharp/DuplicateCodeVulnerable.cs
/**
 * Test Fixture: Duplicate Code Vulnerable Pattern (C#)
 *
 * This file contains duplicated code blocks.
 * Expected detections: >=1 violation for AP-009
 * Rule ID: AP-009
 * Severity: HIGH (warning)
 */

using System;

namespace TestFixtures.AntiPatterns
{
    public class DuplicateCodeVulnerable
    {
        // VULNERABLE: Duplicated validation pattern
        public void ProcessUserA(dynamic user)
        {
            if (user == null)
                throw new ArgumentNullException(nameof(user));
            var id = user.Id;
            var name = user.Name;
            var email = user.Email;
            // Process user A
        }

        // VULNERABLE: Same pattern duplicated
        public void ProcessUserB(dynamic user)
        {
            if (user == null)
                throw new ArgumentNullException(nameof(user));
            var id = user.Id;
            var name = user.Name;
            var email = user.Email;
            // Process user B
        }
    }
}

File: tests/fixtures/anti-patterns/typescript/duplicate-code-vulnerable.ts
/**
 * Test Fixture: Duplicate Code Vulnerable Pattern (TypeScript)
 *
 * This file contains duplicated code blocks.
 * Expected detections: >=1 violation for AP-009
 * Rule ID: AP-009
 * Severity: HIGH (warning)
 */

interface User {
  id: number;
  name: string;
  email: string;
}

// VULNERABLE: Duplicated validation pattern
export function processUserA(user: User | null): void {
  if (!user) {
    throw new Error("User is required");
  }
  const id = user.id;
  const name = user.name;
  const email = user.email;
  // Process user A
}

// VULNERABLE: Same pattern duplicated
export function processUserB(user: User | null): void {
  if (!user) {
    throw new Error("User is required");
  }
  const id = user.id;
  const name = user.name;
  const email = user.email;
  // Process user B
}
